service/csvService.go

```diff
@@ -0,0 +1,138 @@
+package service
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/fatih/structs"
+)
+
+type CsvService interface {
+	FilterById(string) ([]Doc, error)
+	DownloadCsvData() error
+}
+
+type CsvServiceImpl struct{}
+
+type SearchResponse struct {
+	Doc      []Doc `json:"docs"`
+	NumFound int   `json:"numFound"`
+	Start    int   `json:"start"`
+}
+
+type Doc struct {
+	Key       string `json:"key"`
+	Title     string `json:"title"`
+	Type      string `json:"type"`
+	Published string `json:"first_published_year"`
+}
+
+var CSVFile = "/home/omar/Desktop/saved.csv"
+
+// func (c )LoadCsvData
+func (c CsvServiceImpl) DownloadCsvData() error {
+	getAndSaveData()
+	return nil
+}
+
+func (c CsvServiceImpl) FilterById(id string) ([]Doc, error) {
+	file, err := os.Open(CSVFile)
+	if err != nil {
+		return nil, err
+	}
+	reader := csv.NewReader(file)
+	records, errReader := reader.ReadAll()
+	if errReader != nil {
+		return nil, errReader
+	}
+
+	var recordsStruct []Doc
+
+	for _, record := range records {
+		fmt.Println(record)
+
+		doc := Doc{
+			Key:       record[0],
+			Title:     record[1],
+			Type:      record[2],
+			Published: record[3],
+		}
+		if doc.Key == id {
+			recordsStruct = append(recordsStruct, doc)
+		}
```

Comment:
This logic results in doing work that you are not sure is needed.

I'd recommend to first check if record[0] == id, and if so, then append the corresponding doc.

```suggestion
		if record[0] == id {
			doc := Doc{
				Key:       record[0],
				Title:     record[1],
				Type:      record[2],
				Published: record[3],
			}
			
			recordsStruct = append(recordsStruct, doc)
		}

```

Reply:
:+1: 
thanks!